docs/ldml/tr35-keyboards.md

```diff
@@ -308,6 +308,269 @@ For extensibility, the `<special>` element will be allowed at nearly every level
 
 See [Element special](tr35.md#special) in Part 1.
 
+## Normalization
+
+Normalization will not typically be the responsibility of the keyboard author, rather this will be managed by the keyboard implementation.
+The keyboard implementation will apply normalization as appropriate when matching transform rules and `<display>` value matching.
+Output from the keyboard, following application of all transform rules, will be normalized to the appropriate form for the keyboard implementation.
```

Comment:
'for' ? or 'by' ?

Reply:
_Which_ form is up to that implementation. But, "appropriate" conveys that.

Reply:
```suggestion
Output from the keyboard, following application of all transform rules, will be normalized to the appropriate form by the keyboard implementation.
```